### ISSUE TITLE
Response body is incorrect

### ISSUE THREAD

**Issue by andersoxie** [davido/gerrit-oauth-provider#76]
I get this message in the  error_log when I try tio log in with a newly created Gerrit server:

[2016-11-24 20:51:49,607] [HTTP-63] ERROR com.google.gerrit.pgm.http.jetty.HiddenErrorHandler : Error in GET /oauth?code=3bec265c28d40387c511&state=OdJ9FBFmokV5ZpKEnHtUQMgsAGP4o1klPV_VSGyHXa8
org.scribe.exceptions.OAuthException: Response body is incorrect. Can't extract a token from this: 'error=incorrect_client_credentials&error_description=The+client_id+and%2For+client_secret+passed+are+incorrect.&error_uri=https%3A%2F%2Fdeveloper.github.c
om%2Fv3%2Foauth%2F%23incorrect-client-credentials'

I use Github oauth and Gerrit 2.13.2

I have an old gerrit server also set up with Github oauh and I vcan not see any differences that would give this impact.

**Comment by davido** [comment]
The+client_id+and%2For+client_secret+passed+are+incorrect

**Comment by andersoxie** [comment]
Can the error message be interpreted as the user does not exists? I have not added that user to the Gerrit server yet. If so, how do I add the first user if I can not login?

I can not see anything wrong with the client it and client secret.

**Comment by andersoxie** [comment]
After the message in the error_log I also get this error stack:

        at org.scribe.extractors.TokenExtractor20Impl.extract(TokenExtractor20Impl.java:32)
        at org.scribe.oauth.OAuth20ServiceImpl.getAccessToken(OAuth20ServiceImpl.java:37)
        at com.googlesource.gerrit.plugins.oauth.GitHubOAuthService.getAccessToken(GitHubOAuthService.java:116)
        at com.google.gerrit.httpd.auth.openid.OAuthSessionOverOpenID.login(OAuthSessionOverOpenID.java:100)
        at com.google.gerrit.httpd.auth.openid.OAuthWebFilterOverOpenID.doFilter(OAuthWebFilterOverOpenID.java:79)
        at com.google.gwtexpui.server.CacheControlFilter.doFilter(CacheControlFilter.java:73)
        at com.google.gerrit.httpd.RunAsFilter.doFilter(RunAsFilter.java:122)
        at com.google.gerrit.httpd.RequestMetricsFilter.doFilter(RequestMetricsFilter.java:60)
        at com.google.gerrit.httpd.AllRequestFilter$FilterProxy$1.doFilter(AllRequestFilter.java:136)
        at com.google.gerrit.httpd.AllRequestFilter$FilterProxy.doFilter(AllRequestFilter.java:105)
        at com.google.gerrit.httpd.RequestContextFilter.doFilter(RequestContextFilter.java:75)
        at com.google.inject.servlet.ManagedFilterPipeline.dispatch(ManagedFilterPipeline.java:120)
        at com.google.inject.servlet.GuiceFilter.doFilter(GuiceFilter.java:135)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:221)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
        at org.eclipse.jetty.server.handler.RequestLogHandler.handle(RequestLogHandler.java:95)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
        at org.eclipse.jetty.server.Server.handle(Server.java:499)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:311)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
        at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:544)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
        at java.lang.Thread.run(Thread.java:745)

**Comment by davido** [comment]
> Can the error message be interpreted as the user does not exists? I have not added that user to the Gerrit server yet. If so, how do I add the first user if I can not login?

The error has nothing to do wit hthe gerrit user, but only with client id and client secret, you configured in gerrit plugin.

 > I can not see anything wrong with the client it and client secret.

Where have yon configured them? There were some issues reported when this data was configured in `secure.config` and wasn't discovred correctly. Have you tried to move them in `gerrit.config`?

**Comment by andersoxie** [comment]
I have them in etc/gerrit.config

Below this row at the end of the file
`[plugin "gerrit-oauth-provider-github-oauth"]`

**Comment by davido** [comment]
> I have them in etc/gerrit.config

This should be fine. So, have you tried to re-generate them again? The only idee what I have is to increase the log level or even set up devn environment and debug the problem.

**Comment by andersoxie** [comment]
I gave it a new try. I copied all files from my old server to the new one, Did some reconfiguration due to adapt to the new server and did get the same problem, so it was not related to that to a specific version of Gerrit. 
Then I tried with reusing the old Github app and then it worked. So something is wrong with my new github app. I will not investigate that further.
Next step will be to update to the latest version of Gerrit but that I will take later when I have more time,
